src/frontend/flight/sync/writer.rs

```diff
@@ -853,19 +854,20 @@ mod tests {
             }
 
             let log_store = ctx.internal_object_store.get_log_store(table_name);
+	    let origin = (*origin).to_string();
 
             // Determine whether this is the last sync of the sequence, i.e. are there no upcoming
             // syncs with the same sequence number from this origin?
             let last = !table_sequence
                 .iter()
                 .skip(sync_no + 1)
-                .any(|&(_, o, s)| *sequence == s && *origin == o);
+                .any(|&(_, o, s)| *sequence == s && o == o);
```

Comment:
Now it's comparing `o` to `o` which is always going to be true, I think we're trying to compare the `o` from the outer loop above to the `o` in the lambda

```suggestion
                .any(|&(_, this_o, s)| *sequence == s && o == this_o);
```

Reply:
Yup, whoops, thank you.